xbmc/peripherals/devices/Peripheral.cpp

```diff
@@ -149,10 +150,21 @@ bool CPeripheral::Initialise(void)
     return bReturn;
 
   g_peripherals.GetSettingsFromMapping(*this);
-  m_strSettingsFile = StringUtils::Format("special://profile/peripheral_data/%s_%s_%s.xml",
-                                          PeripheralTypeTranslator::BusTypeToString(m_mappedBusType),
-                                          m_strVendorId.c_str(),
-                                          m_strProductId.c_str());
+  if (m_iVendorId == 0x0000 && m_iProductId == 0x0000)
+  {
+    m_strSettingsFile = StringUtils::Format("special://profile/peripheral_data/%s_%s.xml",
+                                            PeripheralTypeTranslator::BusTypeToString(m_mappedBusType),
+                                            CUtil::MakeLegalFileName(m_strDeviceName, LEGAL_WIN32_COMPAT).c_str());
+  }
+  else
+  {
+    m_strSettingsFile = StringUtils::Format("special://profile/peripheral_data/%s_%s_%s_%s.xml",
+                                            PeripheralTypeTranslator::BusTypeToString(m_mappedBusType),
+                                            m_strVendorId.c_str(),
+                                            m_strProductId.c_str(),
+                                            CUtil::MakeLegalFileName(m_strDeviceName, LEGAL_WIN32_COMPAT).c_str());
```

Comment:
Thanks @garbear.

http://sprunge.us/aRSA

What are the plans for migration, is this going to happen in Kodi 17? If not I'll submit a PR, otherwise I'll just temporarily add the patch to my LE builds.